src/kedro_neptune/__init__.py

```diff
@@ -217,30 +217,59 @@ def _connection_mode(enabled: bool) -> str:
 
 
 class NeptuneRunDataSet(AbstractDataSet):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._run: Optional[neptune.Run] = None
+        self._loaded: bool = False
+
     def _save(self, data: Dict[str, Any]) -> None:
-        raise NotImplementedError()
+        if self._run is not None:
+            self._run.sync(wait=True)
 
     def _describe(self) -> Dict[str, Any]:
         return {}
 
     def _exists(self) -> bool:
-        return True
-
-    def _load(self) -> Handler:
-        config = get_neptune_config(settings)
-
-        run = neptune.init(
-            api_token=config.api_token,
-            project=config.project,
-            mode=_connection_mode(config.enabled),
+        return self._run is not None
+
+    def __setstate__(self, state):
+        self.__dict__ = state
+        if self._loaded:
+            self._set_run()
+
+    def __getstate__(self) -> dict:
```

Comment:
It's for model serialization right?

Reply:
We have a "fake" dataset for handling logging to Neptune Run. In case of pickling it (for instance in case of multiprocessing) a Run shouldn't be a part of underlying representation and should be created on every unpickle operation.